packages/chain-bridge/src/adapters/StargateAdapter.sol

```diff
@@ -0,0 +1,980 @@
+        external
+        view
+        returns (
+            OFTLimit memory limit,
+            OFTFeeDetail[] memory oftFeeDetails,
+            OFTReceipt memory oftReceipt
+        );
+
+    function stargateType() external pure returns (StargateType);
+
+    function token() external view returns (address);
+}
+
+/**
+ * @title OftCmdHelper
+ * @notice Helper for creating OFT commands for taxi/bus modes
+ */
+library OftCmdHelper {
+    function taxi() internal pure returns (bytes memory) {
+        return "";
+    }
+
+    function bus() internal pure returns (bytes memory) {
+        return new bytes(1);
+    }
+
+    function drive(
+        bytes memory _passengers
+    ) internal pure returns (bytes memory) {
+        return _passengers;
+    }
+}
+
+/**
+ * @title StargateAdapter
+ * @notice Adapter for Stargate V2 Protocol - all V2 contracts are OFT-enabled
+ * @dev Implements IBridgeAdapter interface and connects to Stargate V2 for efficient cross-chain transfers
+ */
+contract StargateAdapter is Ownable, IBridgeAdapter, ILayerZeroComposer {
+    using SafeERC20 for IERC20;
+    using AddressCast for address;
+    using OptionsBuilder for bytes;
+
+    /// @notice Error for unsupported asset
+    error UnsupportedAsset();
+
+    /// @notice Error for insufficient balance
+    error InsufficientBalance();
+
+    /// @notice Transfer parameters struct to avoid stack too deep
+    struct TransferParams {
+        address stargateContract;
+        uint16 destinationChainId;
+        address asset;
+        address recipient;
+        uint256 amount;
+        address originator;
+        address keeper;
+        bytes32 operationId;
+    }
+
+    /// @notice Maximum gas limit for compose execution
+    uint256 public constant MAX_COMPOSE_GAS = 1000000;
+
+    /// @notice Information about failed compose operations for recovery
+    struct FailedCompose {
+        address asset;
+        uint256 amount;
+        address intendedRecipient;
+        bytes32 operationId;
+        address originator;
+        uint16 sourceChainId;
+        uint256 timestamp;
+        bool isDeposit; // true for deposits to FleetProxy, false for withdrawals to CrossChainArk
+    }
+
+    /// @notice Mapping of operation IDs to failed compose details
+    mapping(bytes32 => FailedCompose) public failedComposes;
+
+    /// @notice Array of failed operation IDs for easier iteration
+    bytes32[] public failedOperationIds;
+
+    /*//////////////////////////////////////////////////////////////
+                            STATE VARIABLES
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice The BridgeRouter that manages this adapter
+    address public bridgeRouter;
+
+    /// @notice LayerZero endpoint for compose functionality
+    address public immutable lzEndpoint;
+
+    /// @notice Mapping of supported chains to their LayerZero Endpoint IDs
+    mapping(uint16 chainId => uint32 endpointId) public chainToEndpointId;
```

Comment:
maybe there LZ maintained registry ? could be used and fallback if need custom config.